tests/Resources/ti.network.httpclient.test.js

```diff
@@ -0,0 +1,596 @@
+			should(result.headers['Content-Type']).startWith('multipart/form-data');
+
+			// check name got added
+			should(result).have.property('form');
+			should(result.form).have.property('name');
+			should(result.form.name).eql(newName);
+
+			// check blob data
+			should(result).have.property('files');
+			should(result.files).have.property('attachment');
+			// image/png (Android), image/png (Windows). Ideally this would match the mimetype/contenttype of the file (which it does for Android/Windows). Let's hope it does on iOS?
+			should(result.files.attachment).eql('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJYAAACWCAYAAAA8AXHiAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAsNJREFUeNrs3b1NI0EYgGG8InQZlEFCdl1AfhGRayCiABKugMvJaIECKMP53hgZ6QJujXwzuzvf97zSyoEt2Z55NLO2/LMZx/FCqt1gCASWwBJYElgCS2BJYAksgSWBJbAElgSWwBJYElgCS2BJYAksgSWBJbAElgSWwBJYElgCS2BJYAksgSWBJbAUusuOHuur6froxoolW6GUdSv8qn057jebzXukSRnH8apcPJZja8VapsPAPx4nAiqwqqxU4XBFQdUzrPtouCZQPYA1U8dzqjC4plCV5/oCFlxQRTh57x1XVFQRXhV2iysyqhCwesQVHVUYWD3hyoAqFKwecGVBFQ7WmnGV+95mQRUS1hpxZUMVFtaacP2F6ioLqtCw1oArK6rwsJbElRlVClhL4MqOKg2sOXFNoPqVBVUqWHPgmkD1Uu77OdNYp/vMeytcJ1A9ZBvnlF+mqI0LKrCq44IKrOq4oAKrOi6owKqOCyqwquOCCqzquKACqwmucuygAqsFrmuowGqBCyqwquF6+8fVv40QWGdVTtZ3X2x/n4X6lRuw5kX1Y+ImW7jAqoEq1A+RgLUiVOWc6w0usGqj2p94KwIusM5DdeKtCLjAOh8VXGA1QwUXWM1QwQVWM1RwgdUMFVxgfaK6q40KruSwyuQeQN22QAVXUlhHVLuWqOBKBmtOVHAlgbUEKriCw1oSVXZcQzJU+zlRZcY1QAUXWP+PatF/Ys2Ea4AKLrA6R5UJ1wAVXGAFQZUB1wAVXGAFQxUZ1wDV+nGBNV8/I6H6Bi6wZmobDdUJXN11GWAuDsieyvZ4ISuWwJLO2NJtIbJiCSyBJYElsASWBJbAElgSWAJLYElgCSyBJYElsASWBJbAElgSWAJLYElgCSyBJYElsASWBJbAElgSWAJLYElgCSyBJYElsASWBJbAElgSWFqyPwIMAMpfdKkmd/FSAAAAAElFTkSuQmCC');
+
+			finish();
+		};
+		xhr.onerror = function (e) {
+			if (attempts-- > 0) {
+				Ti.API.warn('failed, attempting to retry request...');
+				xhr.send();
+			} else {
+				Ti.API.debug(JSON.stringify(e, null, 2));
+				finish(new Error(e.error || this.responseText));
+			}
+		};
+
+		xhr.open('POST', 'http://www.httpbin.org/post');
+
+		form = {
+			name: newName,
+			attachment: blob
+		};
+
+		xhr.send(form);
+	});
+
+	it.android('HttpResponseCache xhr.cache default', function (finish) {
+		var cache = Ti.Android.HttpResponseCache,
+			url = 'http://www.httpbin.org/etag/default' + Math.random(),
+			xhr = Ti.Network.createHTTPClient(),
+			attempts = 3;
+
+		cache.setHttpCachePath('path_4');
+		cache.setHttpCacheSize(30 * 1024 * 1024);
+		cache.install();
+
+		xhr.setTimeout(6e4);
+		xhr.onload = function () {
+			should(cache.getRequestCount()).be.eql(0);
+			should(cache.getHitCount()).be.eql(0);
+			finish();
+		};
+		xhr.onerror = function (e) {
+			if (attempts-- > 0) {
+				Ti.API.warn('failed, attempting to retry request...');
+				xhr.send();
+			} else {
+				Ti.API.debug(JSON.stringify(e, null, 2));
+				finish(new Error('failed to "' + url + '": ' + e));
+			}
+		};
+
+		xhr.open('GET', url);
+		xhr.send();
+	});
+
+	it.android('HttpResponseCache xhr.cache custom', function (finish) {
```

Comment:
Again, you can just add an "add-on test" to add new unit tests to the suite and not have to copy/override the existing API test file. Makes it easier for you and for us to see the new tests added.

Copying and modifying is what's necessary for modifying existing tests (say fixing a test we skip on a given platform and re-enabling it).

So anyways, for next time, you could just create a `ti.network.httpclient.addontest.js` and add these two new tests to it.

Reply:
done